PeerioTechnologies/peerio-desktop
# feat: UI feedback for poor connection when downloading URL image

[PR guideline](https://github.com/PeerioTechnologies/peerio-icebear/blob/dev/docs/CONTRIBUTING.md)

### PR author

#### Relevant issue tracker links  
https://peerio.tpondemand.com/entity/7654
#### Linked PRs in other repos
n/a
#### Manual testing instructions  
"Poor connection..." message should appear when client app is attempting to download URL image preview, but the app is offline. It should not appear if: image is not supposed to download (e.g. oversize or user has disabled image previews), or if user is online.
#### Additional notes
Not related to this particular task, but I'm finding that the live retrying of images downloads isn't working. I need to click to a different chat and then come back.

----
### Reviewer

* Localisation.
* Architectural decisions.
* Performance.
* Comments.
* Big feature tech design documentation.
* Readability.
* DRY.
* Can we extract something to SDK?
* Are there any bicycles invented that should be replaced with existing SDK
* Are there unit and e2e tests added?

----
### Repository owner
